Make test more realistic (use the fields) and add test using Boost's tokenizer

daveisfera opened the pull request:
The existing version of the test didn't actually "use" the result, so split7.cpp was basically just counting tokens instead of making them available. This "uses" the data by counting the number of words and characters to verify that it's correct and that the data is actually being generated. This change has some interesting impacts as follows:
1. Python results are slower than C/C++ results (it appears that the individual strings weren't be "created" or something like that)
2. Version based on StrTk (split7) is more in line with the StringRef version (split6) like you'd expect
3. Version based on Boost's tokenizer is notoriously slow when "using" the data and was almost as fast as the StrTk version, but "using" the data reveals that it's the slowest C/C++ version and comparable to the slower Python versions.

daveisfera committed:
Adding split8 that uses Boost's Tokenizer

daveisfera committed:
Getting it to build with C++98/03

daveisfera committed:
Replacing tabs with spaces

daveisfera committed:
Adding counts of the number of words and characters to make the test more realistic

tobbez commented:
This show why you should be careful with micro-benchmarks. The results probably become a bit more generally meaningful with these changes.

Content-wise this looks fine, but I'd appreciate if you could update the commit messages to follow [these guidelines](https://gist.githubusercontent.com/tobbez/55686ef79bf0b593d90a/raw/789e64318fd91ac598eb814320cec3f25543a7c3/good-commit-messages.txt), i.e. use verbs in the imperative, and reword the last one to make the header line short enough, maybe something along the lines of:

```
Add counting of the number of words and characters

This makes the benchmarks more realistic.
```

Thanks.

On a side note, any idea what sparked the recent interest in this repository? It has gone unnoticed for several year, but now there's been multiple pull requests and stars over the last few days.

daveisfera commented:
I'll make the requested changes to the commit messages. I believe the interest came from [this reddit post](http://www.reddit.com/r/cpp/comments/326rww/string_splitting_c_vs_python/) or at least that's what drew my attention to it.

daveisfera commented:
https://help.github.com/articles/changing-a-commit-message/#amending-older-or-multiple-commit-messages
The above link sounds like changing commit messages is discouraged, so I would prefer not to do that if it's ok.

daveisfera committed:
Updated version using StrTk from Arash Partow

daveisfera committed:
Adding version that uses boost::algorithm::split()

daveisfera committed:
Minor cleanup

daveisfera committed:
Improving performance with an output parameter instead of returning a new vector each time

tobbez commented:
It's generally advised not to change the commit history when other people may have cloned (and depend on) the repository, but we can assume that this is not an issue in this case. Rewriting history and force pushing is done quite commonly in the context of pull requests.

daveisfera committed:
Getting rid of tabs

daveisfera committed:
Improving measurement accuracy
Using clock_gettime() to measure elapsed time at better than second precision
Printing a single decimal place of floating point output to give some context without overwhelming the user/output

daveisfera commented:
I'm not very familiar with git (I always just use hg-git) and after trying to figure out how to edit the old messages and googling not turning up anything but "use command so and so" and/or "it's painful", I gave up. Honestly, I don't care enough to go back and fix the commit messages. If you'd like to pull this as is or fix them yourself, then I'd be glad to have these changes included but my interest level isn't high enough to figure out git so I could clean up the commit log.

tobbez commented:
I've cleaned up the commit history and merged these changes. Thanks.
